### PR TITLE
Added ability to block inbound connections by IP

### PR DESCRIPTION
**Pull request by oxtoacart**
For getlantern/http-proxy-lantern#55

Depends on getlantern/tlsdefaults#3

### CODE REVIEW

File: http_proxy_test.go [hunk 1]
```diff
@@ -196,7 +197,7 @@ func impatientProxy(maxConns uint64, idleTimeout time.Duration) (string, error)
 	}
 	go func(err *error) {
 		if *err = srv.ListenAndServeHTTP("localhost:0", wait); err != nil {
-			log.Errorf("Unable to serve: %s", err)
+			log.Errorf("Unable to serve: %v", err)
```
Comment (oxtoacart) [inline http_proxy_test.go]: Linter issue

File: http_proxy_test.go [hunk 2]
```diff
@@ -312,7 +313,7 @@ func TestConnectOK(t *testing.T) {
 		}
 
 		buf = [400]byte{}
-		_, err = conn.Read(buf[:])
```
Comment (oxtoacart) [inline http_proxy_test.go]: Linter issue (same below)

File: http_proxy_test.go [hunk 3]
```diff
@@ -515,7 +541,7 @@ func setupNewHTTPServer(maxConns uint64, idleTimeout time.Duration) (string, err
 	}
 	go func(err *error) {
 		if *err = s.ListenAndServeHTTP("localhost:0", wait); err != nil {
-			log.Errorf("Unable to serve: %s", err)
+			log.Errorf("Unable to serve: %v", err)
```
Comment (oxtoacart) [inline http_proxy_test.go]: Linter issue

**Comment by myleshorton** [comment]
Odd that CI is failing here @oxtoacart -- failing on `server/server.go:92: undefined: tlsdefaults.NewListener` You pushed those tlsdefaults changes upstream right? or are those in their own PR?

**Comment by oxtoacart** [comment]
@myleshorton They're on their own PR

**Comment by oxtoacart** [comment]
Given that our root cause is actually on the client, I'm on the fence as to whether or not to even leave these PR's open.  What do you think @myleshorton?

**Comment by myleshorton** [comment]
Yeah good question @oxtoacart. Maybe if it ain't broke don't fix it?

**Comment by oxtoacart** [comment]
@myleshorton That's my feeling too.  Closing these.

**Comment by aranhoide** [comment]
LGTM.

**Pull request merged**